src/main/java/com/ticketbookingsys/metro/aspect/StationValidationAspect.java

```diff
@@ -0,0 +1,52 @@
+package com.ticketbookingsys.metro.aspect;
+
+import com.ticketbookingsys.metro.entity.STATION_TYPE;
+import lombok.extern.slf4j.Slf4j;
+import org.aspectj.lang.JoinPoint;
+import org.aspectj.lang.annotation.Around;
+import org.aspectj.lang.annotation.Aspect;
+import org.aspectj.lang.annotation.Before;
+import org.springframework.stereotype.Component;
+
+import java.lang.reflect.Field;
+
+@Slf4j
+@Aspect
+@Component
+public class StationValidationAspect {
+
+    /*
+     * TODO : price restrictions on Stations
+     *  accept a new parameter stationType
+     *  if stationType are of this category (LUX, SEMI-LUX) then price should be minimum 20
+     *   if stationType are of this category (NON-LUX) then price should be maximum 50
+     * */
+
+    @Before(value = "@annotation(com.ticketbookingsys.metro.annotation.StationValidation)")
+    public void beforeAdvice(JoinPoint joinPoint) throws Throwable {
+        Object[] args = joinPoint.getArgs();
+        for (Object object : args) {
+            if (object != null) {
+                STATION_TYPE stationType = null;
+                Long price = null;
+                for (Field field : object.getClass().getDeclaredFields()) {
```

Comment:
use lambda for

Reply:
Did some modifications, but i think something more can be done .. can you suggest some? there was some issue with Internal for loop using lamda.